src/core/CorePool.ts

```diff
@@ -311,11 +343,63 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): { position: Position; amount0: JSBI; amount1: JSBI } {
-    // TODO
+    let amount0: JSBI = ZERO,
+      amount1: JSBI = ZERO;
+
+    let position: Position = this.getPosition(owner, tickLower, tickUpper);
+
+    // use assert
+    if (tickLower > tickUpper) {
+      // error handling
+      console.error("[Error]: tickLower upper than tickUpper");
+    } else if (tickLower < TickMath.MIN_TICK) {
+      console.error("[Error]: tickLower lower than MIN_TICK");
+    } else if (tickUpper > TickMath.MAX_TICK) {
+      console.error("[Error]: tickUpper bigger than MAX_TICK");
+    } else {
+      // check ticks pass, update position
+      this.updatePosition(owner, tickLower, tickUpper, liquidityDelta);
```

Comment:
It's ok to save and return position reference from the return value here.

Reply:
Make sense, we don't need to get position twice